test/test_svd.cc

```diff
@@ -270,92 +341,42 @@ void test_svd_work( Params& params, bool run )
             // Run ScaLAPACK reference routine.
             //==================================================
             double time = barrier_get_wtime(MPI_COMM_WORLD);
-            scalapack_pgesvd(job2str(slate::Job::NoVec), job2str(slate::Job::NoVec), m, n,
-                             &Aref_data[0],  1, 1, A_desc, &Sigma_ref[0],
-                             &U_data[0],  1, 1, U_desc,
-                             &VT_data[0], 1, 1, VT_desc,
-                             &work[0], lwork, &rwork[0], &info_ref);
+            scalapack_pgesvd(
+                jobu_str, jobvt_str, m, n,
+                &Aref_data[0],  1, 1, A_desc, &Sigma_ref[0],
+                &U_data[0],  1, 1, U_desc,
+                &VT_data[0], 1, 1, VT_desc,
+                &work[0], lwork, &rwork[0], &info_ref );
             slate_assert(info_ref == 0);
             time = barrier_get_wtime(MPI_COMM_WORLD) - time;
 
             params.ref_time() = time;
 
-            //==================================================
-            // Test results by checking relative forward error
-            //
-            //      || Sigma_ref - Sigma ||
-            //     ------------------------- < tol * epsilon
-            //         || Sigma_ref ||
-            //==================================================
-            real_t Sigma_ref_norm = blas::asum(Sigma_ref.size(), &Sigma_ref[0], 1);
-            // Perform a local operation to get differences Sigma = Sigma - Sigma_ref
-            blas::axpy(Sigma_ref.size(), -1.0, &Sigma[0], 1, &Sigma_ref[0], 1);
+            if (! ref_only) {
+                //==================================================
+                // Test results by checking relative forward error
+                //
+                //      || Sigma_ref - Sigma ||
+                //     ------------------------- < tol * epsilon
+                //         || Sigma_ref ||
+                //==================================================
+                real_t Sigma_ref_norm = blas::asum( min_mn, &Sigma_ref[0], 1 );
+                // Perform a local operation to get differences Sigma = Sigma - Sigma_ref
+                blas::axpy( min_mn, -1.0, &Sigma[0], 1, &Sigma_ref[0], 1 );
 
-            params.error() = blas::asum(Sigma.size(), &Sigma_ref[0], 1)
-                           / Sigma_ref_norm;
+                params.error() = blas::asum( min_mn, &Sigma_ref[0], 1 )
+                               / Sigma_ref_norm;
+
+                params.okay() = (params.error() <= tol);
```

Comment:
This should be changed to `params.okay() = params.okay() && (params.error() <= tol);` since it's no longer the first check.

Reply:
Good catch.

Reply:
Fixed.